### PR TITLE
BugFix : Fixed Improper Handling for Non-Numeric Values in Allotted hours section of Create Action item modal

### LINKED ISSUE #3368: Bug Report : Improper input handling for Allotted hours in the Create Action Item modal
**Describe the bug**
When entering non-numeric characters in the allotted hours input field, the field value becomes NaN (Not a Number) and remains stuck in this state. Once NaN appears, the field cannot be cleared or updated with new values, even when attempting to backspace or enter valid numbers. This creates a blocking user experience where the only way to recover is to refresh the form.
To Reproduce

**Steps to reproduce the behavior:**
1. Login as a admin or superadmin
2. Navigate to action items 
3. Try to create a action item
4. type a letter in the allotted hours section.

**Expected behavior**

1.When non-numeric characters are entered, the field should either prevent the input or maintain its previous valid value
2.Users should be able to clear the field and enter new values at any time
3.The field should never display NaN to end users
4.Invalid inputs should be handled gracefully without getting the field stuck in an unrecoverable state

**Actual behavior**

1.Entering a letter or non-numeric character causes the field value to become NaN
2.Once NaN appears, the field becomes unresponsive to new inputs
3.Backspace and delete operations fail to clear the NaN value
4.The field remains stuck in an invalid state until form refresh

**Screenshots**

Image

**Additional details**
Root cause appears to be in the handleFormChange function where NaN values are being stored in the form state without proper validation. The current implementation doesn't properly sanitize numeric inputs before updating the state, leading to this persistent error state.

I have resolved this error on  my local machine and am ready to raise a PR for this.

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by yugal07**
**What kind of change does this PR introduce?**

BugFix

**Issue Number:**

Fixes #3368 

**Snapshots/Videos:**

Before 

https://github.com/user-attachments/assets/4fd6a4ce-ef56-4a21-a1a2-c8729ed7a9c1

After

https://github.com/user-attachments/assets/486f11f1-336a-48c6-a544-3f4bdd48d9c0

**If relevant, did you update the documentation?**

N/A

**Summary**

Added separate handling for allotted hours section, for non numeric values and negative values

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced validation for allotted hours input to ensure only valid, non-negative numbers are accepted.
	- Added test cases to validate that the allotted hours field does not accept letters or negative values.
	- Introduced additional test cases to cover edge cases for allotted hours input, including handling of empty, null, and whitespace-only values.
	- Verified that category selection and assignee type changes are validated correctly.
	- Improved handling of timezone edge cases around Daylight Saving Time changes.
	- Added tests for handling of infinite values, long text inputs, and rapid form field changes.

- **Chores**
	- Removed unnecessary comments in the code to improve clarity and maintainability.

### CODE REVIEW

**Comment by yugal07** [comment]
@palisadoes, could this be a flaky test? As far as I know, a CSS import isn't needed in a test file. If possible, could you re-run the tests? Thanks!

**Comment by palisadoes** [comment]
1. It's been fixed.
1. Please make a minor commit to rerun the GitHub action

**Comment by yugal07** [comment]
Hello @palisadoes, test is failing because a file is not in prettier format. However, i did not work in that file. Should i fix it?

**Comment by palisadoes** [comment]
1. Merge your code with the latest upstream.
2. There have been many recently merged PRs. That should fix it

**Comment by palisadoes** [comment]
1. Please make sure coderabbit.ai approves your PR. 
2. It has many suggestions

File: src/screens/OrganizationActionItems/ItemModal.spec.tsx [hunk 1]
```diff
@@ -802,6 +1154,267 @@ describe('Testing ItemModal', () => {
...
+    fireEvent.change(dateInput, { target: { value: futureDate } });
+    expect(dateInput).toHaveValue(futureDate);
+
+    // Test form submission with valid date
+    const submitButton = screen.getByTestId('submitBtn');
+    fireEvent.click(submitButton);
+    await waitFor(() => {
+      expect(toast.success).toHaveBeenCalledWith(t.successfulCreation);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (yugal07) [reply src/screens/OrganizationActionItems/ItemModal.spec.tsx:1351]: MUI DatePicker doesn't handle invalid dates the way we're testing. When an invalid date is entered:

It doesn't clear the input (value='')
It maintains the last valid date value
It handles validation internally without exposing error messages directly

**Comment by yugal07** [comment]
@palisadoes, the remaining suggestions from Coderabbits are invalid because the application utilizes MUI components, which are highly versatile and support various input types. I have already implemented the valid suggestions, Let me know if there's need for further adjustments.

File: src/screens/OrganizationActionItems/ItemModal.spec.tsx [hunk 2]
```diff
@@ -790,6 +899,249 @@
...
+    expect(preCompletionNotes).toHaveValue('∑ π ∆ ∞ € £ ¥ ₹ ± ≠ ≈ ∴ ∵');
+
+    const submitButton = screen.getByTestId('submitBtn');
+    fireEvent.click(submitButton);
+    await waitFor(() => {
+      expect(toast.success).toHaveBeenCalledWith(t.successfulCreation);
+      // Verify the last special character value persists after submission
+      expect(preCompletionNotes).toHaveValue('∑ π ∆ ∞ € £ ¥ ₹ ± ≠ ≈ ∴ ∵');
+    });
+  });
```
[the remark this answers is not in the document]
Comment (yugal07) [reply src/screens/OrganizationActionItems/ItemModal.spec.tsx:1058]: Done

[the remark this answers is not in the document]
Comment (yugal07) [reply src/screens/OrganizationActionItems/ItemModal.spec.tsx:1058]: in a different test

**Comment by palisadoes** [comment]
1. We have to close this.
1. We have been having issues merging PRs since migrating from Admin using a MongoDB backend to PostgreSQL
1. Please create a new branch based on the latest upstream and resubmit your PR
